compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2943,7 +2956,7 @@ private void reorderNamedArgs(BLangInvocation iExpr, BInvokableSymbol invokableS
             if (namedArgs.containsKey(param.name.value)) {
                 expr = namedArgs.get(param.name.value);
             } else {
-                expr = getDefaultValueLiteral(param.defaultValue);
+                expr = getDefaultValueLiteral(param.defaultValue, param.type.tag);
```

Comment:
please fix this properly in a different PR

Reply:
PR https://github.com/ballerina-platform/ballerina-lang/pull/11383 fixes this.